File: ios/AgreeWise/AgreeWise/ML/ScoreCalculator.swift
import Foundation

enum ScoreCalculator {
    static func calculateScore(text: String) -> Int {
        var score = 0
        let textToScan = text.lowercased()
        for (keyword, weight) in KeywordDatabase.keywordWeights {
            if textToScan.contains(keyword) { score += weight }
        }
        return score
    }
}

File: app/src/main/java/com/example/agreewise/ml/ScoreCalculator.kt
package com.example.agreewise.ml

object ScoreCalculator {

    fun calculateScore(text: String): Int {
        var totalScore = 0
        val textToScan = text.lowercase()

        // Sort keywords by length descending to handle "limitation of liability" before "liability"
        val sortedKeywords = KeywordDatabase.keywordWeights.keys.sortedByDescending { it.length }
        val foundKeywords = mutableSetOf<String>()

        var remainingText = textToScan
        for (keyword in sortedKeywords) {
            if (remainingText.contains(keyword)) {
                totalScore += KeywordDatabase.keywordWeights[keyword] ?: 0
                foundKeywords.add(keyword)
                // Remove the keyword from text so it's not double-counted by shorter substrings
                remainingText = remainingText.replace(keyword, " ")
            }
        }

        return totalScore
    }
}